### PR TITLE
Remove things from base/

### PR DESCRIPTION
**Pull request by howardjohn**
This completes the work done in https://github.com/istio/istio/pull/31790 to migrate these from base/ to istiod/. This was first done in 1.11. The removal was not done as a matter of backwards compatibility so folks could migrate to the new ServiceAccount, etc.

 I believe it is reasonable to complete the move now.

### CODE REVIEW

**Comment by howardjohn** [comment]
stop

**Review by zirain — APPROVED** [review]
this LGTM, but want more eyes on this.

@istio/wg-environments-maintainers PTAL

**Review by hanxiaop — APPROVED** [review]

File: releasenotes/notes/manifest-base-cleanup.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+releaseNotes:
+  - |
+    **Removed** obsolete manifests from the `base` Helm chart. See Upgrade Notes for more information.
+upgradeNotes:
+  - title: "`base` Helm Chart removals"
+    content: |
+      A number of configurations previously present in the the `base` Helm chart were *copied* to the `istiod` chart in a previous releases.
```
Comment (linsun) [inline releasenotes/notes/manifest-base-cleanup.yaml:10]: What exactly is the upgrade implication?   Seems for upgrade, users will still see these, but it is safe for them to manually remove these roles?

Comment (howardjohn) [reply releasenotes/notes/manifest-base-cleanup.yaml:10]: They are removed on upgrade

**Review by hzxuzhonghu — APPROVED** [review]

**Review by linsun — APPROVED** [review]
I think this is a good move lgtm

**Pull request merged**